src/z_dateformat.erl

```diff
@@ -304,14 +305,26 @@ tag_to_value($w, Date, _, _Options) ->
 tag_to_value($W, {Y,M,D}, _, _Options) ->
    integer_to_list(year_weeknum(Y,M,D));
 
+% Absolute year with at least 4 digits, padded with zeroes (e.g. '0003')
+tag_to_value($x, {Y, _, _}, _, _Options) when Y >= 1000; Y =< -1000 ->
+    integer_to_list(erlang:abs(Y));
+tag_to_value($x, {Y, _, _}, _, _Options) when Y < 0 ->
+    io_lib:format("~4..0B", [0 - Y]);
+tag_to_value($x, {Y, _, _}, _, _Options) ->
+    io_lib:format("~4..0B", [Y]);
+
 % Year, 2 digits; e.g. '99'
 tag_to_value($y, {Y, _, _}, _, _Options) ->
    Y1 = Y rem 100,
    [ Y1 div 10 + $0, Y1 rem 10 + $0];
 
-% Year, 4 digits; e.g. '1999'
+% Year, (at least) 4 digits; e.g. '1999' or '-4150'
+tag_to_value($Y, {Y, _, _}, _, _Options) when Y >= 1000; Y =< -1000 ->
+    integer_to_list(Y);
+tag_to_value($Y, {Y, _, _}, _, _Options) when Y < 0 ->
+    [$-, io_lib:format("~4..0B", [0 - Y]) ];
 tag_to_value($Y, {Y, _, _}, _, _Options) ->
-   integer_to_list(Y);
+    io_lib:format("~4..0B", [Y]);
```

Comment:
Why change this for `$Y`? In templates you probably want to render: the year 313 (not 0313).

Reply:
The thing is that the specs say *4 digit number*.
Should we follow the specs or something we expect?

Reply:
We should then change the documentation to *full (possibly signed) year*

Reply:
I'll change the code here and the documentation of zotonic.

Reply:
For the ISO specs, we have the 4 digit year, which we produce in `$x`. For template rendering, however, you want the simple year, with `$Y`.